src/CosmosDB/CosmosDB/SQL/GetAzCosmosDBSqlContainer.cs

```diff
@@ -0,0 +1,79 @@
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.CosmosDB.Models;
+using Microsoft.Azure.Commands.CosmosDB.Models;
+using Microsoft.Azure.Commands.CosmosDB.Helpers;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.CosmosDB
+{
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDBSqlContainer", DefaultParameterSetName = NameParameterSet), OutputType(typeof(PSSqlContainerGetResults), typeof(PSThroughputSettingsGetResults))]
+    public class GetAzCosmosDBSqlContainer : AzureCosmosDBCmdletBase
+    {
+        [Parameter(Mandatory = false, ParameterSetName = NameParameterSet, HelpMessage = Constants.AccountNameHelpMessage)]
+        public string AccountName;
+
+        [Parameter(Mandatory = false, ParameterSetName = NameParameterSet, HelpMessage = Constants.DatabaseNameHelpMessage)]
+        public string DatabaseName;
+
+        [Parameter(Mandatory = false, ParameterSetName = NameParameterSet, HelpMessage = Constants.ResourceGroupNameHelpMessage)]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = NameParameterSet, HelpMessage = Constants.ContainerNameHelpMessage)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ParentObjectParameterSet, HelpMessage = Constants.SqlDatabaseObjectHelpMessage )]
+        public PSSqlDatabaseGetResults InputObject{ get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.SqlContainerDetailedParamHelpMessage)]
+        public SwitchParameter Detailed { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if(ParameterSetName.Equals(ParentObjectParameterSet))
+            {
+                ResourceIdentifier resourceIdentifier = new ResourceIdentifier(InputObject.Id);
+                ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                DatabaseName = resourceIdentifier.ResourceName;
+                AccountName = resourceIdentifier.ParentResource;
+            }
+
+            if(!string.IsNullOrEmpty(Name))
+            {
+                SqlContainerGetResults sqlContainerGetResults = CosmosDBManagementClient.SqlResources.GetSqlContainerWithHttpMessagesAsync(ResourceGroupName, AccountName, DatabaseName, Name).GetAwaiter().GetResult().Body;
+                WriteObject(new PSSqlContainerGetResults(sqlContainerGetResults));
+
+                if (Detailed)
+                {
+                    ThroughputSettingsGetResults throughputSettingsGetResults = CosmosDBManagementClient.SqlResources.GetSqlContainerThroughputWithHttpMessagesAsync(ResourceGroupName, AccountName, DatabaseName, Name).GetAwaiter().GetResult().Body;
```

Comment:
These three variables should never be empty, of user doesnt provide these, then an exception from backend of our code is thrown. Do you want me to catch that and translate it into a new exception ?? But because the exception we throw is informative enough, I did not translate it.

Reply:
made them mandatory.